packages/vite-node/src/server.ts

```diff
@@ -392,6 +393,7 @@ export class ViteNodeServer {
     return withInlineSourcemap(result, {
       filepath: mod?.file || filepath,
       root: this.server.config.root,
+      noFirstLineMapping: Number(viteVersion.split('.')[0]) >= 6,
```

Comment:
I'll add `pkg.pr.new` tracking this PR and check if this works in minimal repro with Vite 5 and 6.